### PR TITLE
Change Default CI java version to 11

### LINKED ISSUE #260: Default CI Java Version to Java 11, run tests on 8, 14 and 17
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/64, default CI Java version to 11 (LTS), and run tests on 8, 14 and 17.

- [ ] Change the default Java version for CI in GitHub actions from 14/15 to 11 (LTS).
- [ ] Add Java 8, 14 (used before), and 17 (next LTS) to the CI matrix.
- [ ] Document Java versions in DEVELOPER_GUIDE.md.

### PR DESCRIPTION
**Pull request by davidcui1225**
Signed-off-by: David Cui <davidcui@amazon.com>

### Description
* Change default CI java version to 11 
* Add Java versions 11 and 14 to the CI matrix
* Update Developer guide 

Note: Java 8 and 17 will be added in a later PR after being unblocked to resolve #260 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/opensearch-observability-test-and-build-workflow.yml [hunk 1]
```diff
@@ -9,15 +9,19 @@ env:
 
 jobs:
   build:
+    strategy:
+      matrix:
+        java: [8, 11, 14, 17]
```
Comment (joshuali925) [inline .github/workflows/opensearch-observability-test-and-build-workflow.yml:14]: remove matrix if you are not using it in this workflow?

Comment (davidcui1225) [reply .github/workflows/opensearch-observability-test-and-build-workflow.yml:14]: I'll need to make changes to correctly use the matrix versions but we need to have it as per the release campaign

Comment (joshuali925) [reply .github/workflows/opensearch-observability-test-and-build-workflow.yml:14]: got it. forgot about that i thought it's only to change default version

**Review by ps48 — DISMISSED** [review]

**Review by joshuali925 — APPROVED** [review]

**Review by eugenesk24 — APPROVED** [review]
Good work!

**Pull request merged**